src/OrchardCore.Modules/OrchardCore.DataProtection/Startup.cs

```diff
@@ -0,0 +1,38 @@
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Modules;
+
+namespace OrchardCore.DataProtection
+{
+    public class Startup : StartupBase
+    {
+        private readonly IDataProtectionProvider _dataProtectionProvider;
+        private readonly ShellSettings _shellSettings;
+
+        public Startup(
+            IDataProtectionProvider dataProtectionProvider,
+            ShellSettings shellSettings)
+        {
+            _dataProtectionProvider = dataProtectionProvider;
+            _shellSettings = shellSettings;
+        }
+
+        public override void ConfigureServices(IServiceCollection services)
+        {
+            // Replace the global data protection provider by a tenant-aware instance so that modules that
+            // internally rely on IDataProtector/IDataProtectionProvider automatically and transparently get
+            // an isolated instance that doesn't allow decrypting payloads encrypted by another tenant.
+            // Services that need to retrieve a non-isolated provider can use IHostDataProtectionProvider.
+            services.Replace(ServiceDescriptor.Singleton(typeof(IDataProtectionProvider),
```

Comment:
Just so that users just can still resolve `IDataProtectionProvider` and we handle the isolation transparently?